services/galley/src/Galley/Run.hs

```diff
@@ -0,0 +1,55 @@
+module Galley.Run (run) where
+
+import           Imports
+
+import           Cassandra                          (runClient, shutdown)
+import           Cassandra.Schema                   (versionCheck)
+import           Control.Exception                  (finally)
+import           Control.Lens                       ((^.))
+import           Data.Metrics.Middleware.Prometheus (waiPrometheusMiddleware)
+import           Data.Metrics.WaiRoute              (treeToPaths)
+import           Data.Misc                          (portNumber)
+import           Data.Text                          (unpack)
+import           Network.Wai                        (Middleware)
+import           Network.Wai.Utilities.Server
+import           Util.Options
+import qualified Control.Concurrent.Async           as Async
+import qualified Data.Metrics.Middleware            as M
+import qualified Network.Wai.Middleware.Gunzip      as GZip
+import qualified Network.Wai.Middleware.Gzip        as GZip
+import qualified System.Logger                      as Log
+
+import           Galley.API          (sitemap)
+import qualified Galley.API.Internal as Internal
+import qualified Galley.App          as App
+import Galley.App
+import qualified Galley.Data         as Data
+import           Galley.Options      (Opts, optGalley)
+
+run :: Opts -> IO ()
+run o = do
+    m <- M.metrics
+    e <- App.createEnv m o
+    let l = e ^. App.applog
+    s <- newSettings $ defaultServer (unpack $ o ^. optGalley.epHost)
+                                     (portNumber $ fromIntegral $ o ^. optGalley . epPort)
+                                     l
+                                     m
+    runClient (e^.cstate) $
+        versionCheck Data.schemaVersion
+    d <- Async.async $ evalGalley e Internal.deleteLoop
+    let rtree    = compile sitemap
+        app r k  = runGalley e r (route rtree r k)
+        measured :: Middleware
+        measured = measureRequests m (treeToPaths rtree)
+        middlewares :: Middleware
+        middlewares = waiPrometheusMiddleware sitemap
```

Comment:
I'm assuming this is the only relevant change?  It's a bit hard to see from the diff.

Reply:
Aside from some refactors this is the main thing going on; I wanted a consistent and easy to read middleware setup for each project so I moved a few things around to clean it up :)